### PR TITLE
Add Che preferences API

### PR DESCRIPTION
**Pull request by mmorhun**
Adds methods to use Che client preferences API

Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### CODE REVIEW

**Comment by benoitf** [comment]
hello, FYI there are conflicting files ?

**Comment by mmorhun** [comment]
Yeah, I see, tnx

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts [hunk 1]
```diff
@@ -17,14 +17,38 @@ export class CheApiServiceImpl implements CheApiService {
...
+    }
+
+    async replaceUserPreferences(preferences: Preferences): Promise<Preferences> {
+        const apiClient = await this.getCheApiClient();
+        return await apiClient.replaceUserPreferences(preferences);
+    }
+
+    async deleteUserPreferences(list?: string[]): Promise<void> {
+        const apiClient = await this.getCheApiClient();
+        return await apiClient.deleteUserPreferences(list);
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts]: hello, why do we need this await ?

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts [hunk 2]
```diff
@@ -17,14 +17,38 @@ export class CheApiServiceImpl implements CheApiService {
...
+
     async currentWorkspace(): Promise<cheApi.workspace.Workspace> {
         try {
             const workspaceId = process.env.CHE_WORKSPACE_ID;
             if (!workspaceId) {
                 return Promise.reject('Cannot find Che workspace id, environment variable "CHE_WORKSPACE_ID" is not set');
             }
 
-            const wsClient = await this.wsClient();
+            const wsClient = await this.getCheApiClient();
```
Comment (benoitf) [inline extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts]: probably rename wsClient as well in that case ?

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts (hunk 1 above)
Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts]: https://eslint.org/docs/rules/no-return-await

Comment (mmorhun) [reply extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts]: Good catch

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts (hunk 2 above)
Comment (mmorhun) [reply extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts]: done

**Comment by benoitf** [comment]
seems eclipse che project is not pushing anymore updates on https://www.npmjs.com/package/@eclipse-che/api

**Pull request merged**